src/Command/ArrayGeneratorCommand.php

```diff
@@ -36,42 +36,43 @@ public function __construct(CollectionGeneratorInterface $collectionGenerator)
     protected function configure(): void
     {
         $this->setName('generate:array')
-            ->setDescription('Generate generic array - array<type>()')
-            ->addArgument(
-                'type',
-                InputArgument::REQUIRED,
-                "Type of generic array. It can by bool, int, float, string, array \n" .
-                "or full class namespace (remember to use \\\\ to separate names)."
-            )
-            ->addArgument(
-                'namespace',
-                InputArgument::REQUIRED,
-                "Namespace of new generic array."
-            )
-            ->addOption(
-                'save',
-                's',
-                InputOption::VALUE_OPTIONAL,
-                'Save generated array to generated-collections.json file.',
-                false
-            );
+             ->setDescription('Generate generic array - array<type>()')
+             ->addArgument(
+                 'type',
+                 InputArgument::REQUIRED,
+                 "Type of generic array. It can by bool, int, float, string, array \n" .
+                 "or full class namespace (remember to use \\\\ to separate names)."
+             )
+             ->addArgument(
+                 'namespace',
+                 InputArgument::REQUIRED,
+                 "Namespace of new generic array."
+             )
+             ->addOption(
+                 'save',
+                 's',
+                 InputOption::VALUE_OPTIONAL,
+                 'Save generated array to generated-collections.json file.',
+                 false
+             );
     }
 
     /**
      * @inheritDoc
      */
     protected function execute(InputInterface $input, OutputInterface $output): int
     {
-        $type = $input->getArgument('type');
-        $namespace = $input->getArgument('namespace');
+        $type              = ltrim($input->getArgument('type'),'\\');
+        $namespace         = $input->getArgument('namespace');
         $genericCollection = new GenericCollection($type, $namespace);
-        $save = $input->getOption('save');
+        $save              = $input->getOption('save');
 
-        if ($save !== true && $save !== 'true' && $save !== 'false') {
+        if (!in_array($save, [true, false, 'true', 'false'], true)) {
             throw new \InvalidArgumentException('Possible values for save option are true or false');
         }
-
-        $save = $save === 'false' ? false : true;
+        if (!is_bool($save)) {
+            $save = $save === 'false' ? false : true;
```

Comment:
Maybe `$save = \filter_var($save, \FILTER_VALIDATE_BOOLEAN)`?

Reply:
I'd imagine is_bool is a lot more performant as its such a simple method